src/main/java/org/jfree/chart/entity/EntityCollection.java

```diff
@@ -30,14 +30,15 @@
  * (C) Copyright 2002-2022, by David Gilbert.
  *
  * Original Author:  David Gilbert;
- * Contributor(s):   -;
+ * Contributor(s):   T. Hiltbrand;
  *
  */
 
 package org.jfree.chart.entity;
 
 import java.util.Collection;
 import java.util.Iterator;
+import java.util.Vector;

```

Comment:
It is recommended in the Javadocs for `Vector` (https://docs.oracle.com/javase/8/docs/api/java/util/Vector.html) to use `ArrayList` unless you need synchronisation.

Reply:
Yes, we do need synchronization :-)

Reply:
Instead of burdening all implementations with the synchronization overhead of `Vector`, would [`Collections.synchronizedList()`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#synchronizedList-java.util.List-) be an acceptable alternative for clients?

Reply:
I can do ordered lists with Collections.synchronizedBlahBlahBlah, right?

Reply:
Generally, yes, assuming the backing `Blah` is sorted.